model/map/src/main/java/org/keycloak/models/map/client/MapProtocolMapperUtils.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.models.map.client;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.keycloak.models.ProtocolMapperModel;
+
+public class MapProtocolMapperUtils {
+
+    private final String protocol;
+    private static final Map<String, MapProtocolMapperUtils> INSTANCES = new HashMap<>();
```

Comment:
Please use `ConcurrentMap` as this can be used simultaneously from several threads.